lib/packwerk.rb

```diff
@@ -13,44 +13,84 @@
 module Packwerk
   extend ActiveSupport::Autoload
 
+  # Public APIs
+  autoload :Checker
+  autoload :Cli
+  autoload :Offense
+  autoload :OffensesFormatter
+  autoload :OutputStyle
+  autoload :Package
+  autoload :PackageSet
+  autoload :PackageTodo
+  autoload :Parsers
+  autoload :Reference
+  autoload :ReferenceOffense
+  autoload :Validator
+
+  # Private APIs
+  # Please submit an issue if you have a use-case for these
   autoload :ApplicationLoadPaths
+  private_constant :ApplicationLoadPaths
   autoload :ApplicationValidator
+  private_constant :ApplicationValidator
   autoload :AssociationInspector
-  autoload :OffenseCollection
+  private_constant :AssociationInspector
   autoload :Cache
-  autoload :Checker
-  autoload :Cli
+  private_constant :Cache
   autoload :Configuration
+  private_constant :Configuration
   autoload :ConstantDiscovery
+  private_constant :ConstantDiscovery
   autoload :ConstantNameInspector
+  private_constant :ConstantNameInspector
   autoload :ConstNodeInspector
-  autoload :PackageTodo
+  private_constant :ConstNodeInspector
   autoload :ExtensionLoader
+  private_constant :ExtensionLoader
   autoload :FileProcessor
+  private_constant :FileProcessor
   autoload :FilesForProcessing
+  private_constant :FilesForProcessing
   autoload :Graph
+  private_constant :Graph
   autoload :Loader
+  private_constant :Loader
   autoload :Node
+  private_constant :Node
   autoload :NodeHelpers
+  private_constant :NodeHelpers
   autoload :NodeProcessor
+  private_constant :NodeProcessor
   autoload :NodeProcessorFactory
+  private_constant :NodeProcessorFactory
   autoload :NodeVisitor
-  autoload :Offense
-  autoload :OffensesFormatter
-  autoload :OutputStyle
-  autoload :Package
-  autoload :PackageSet
+  private_constant :NodeVisitor
+  autoload :OffenseCollection
+  private_constant :OffenseCollection
   autoload :ParsedConstantDefinitions
-  autoload :Parsers
+  private_constant :ParsedConstantDefinitions
   autoload :ParseRun
-  autoload :UnresolvedReference
-  autoload :Reference
+  private_constant :ParseRun
```

Comment:
I've always wondered why this is called `ParseRun`. Should we just rename it to `Run`? It is private API so we can do this for free later if need be.

Reply:
Not sure about that one either. Personally, my preference would be to just have API on `packwerk` directly, e.g.
```
Packwerk.check
Packwerk.update_todo
```

This feels like the best UX to me.